### PR TITLE
Push Notifications Test

### PR DESCRIPTION
**Pull request by fabriUCF**

### CODE REVIEW

File: repos/web/src/routes/api/push/+server.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import * as webpush from "web-push";
+import { json } from "@sveltejs/kit";
+
+/** @type {import('./$types').RequestHandler} */
+export async function GET({ url }) {
```
Comment (fabriUCF) [inline repos/web/src/routes/api/push/+server.ts:5]: This needs to be a POST request. Just wanted to make sure this worked

**Comment by rob-3** [comment]
I like the direction this is going! I think instead of an endpoint we probably want a utility function in the `lib/server` folder. The keys also need to be rotated and put in environment variables.

**Comment by rob-3** [comment]
The button actually makes sense for testing rn though

**Comment by rikimbili** [comment]
> I like the direction this is going! I think instead of an endpoint we probably want a utility function in the `lib/server` folder. The keys also need to be rotated and put in environment variables.

I just realized that I've been writing utility and server helper functions outside of lib. I just read the Svelte docs and they recommend using lib for this to keep it organized.

File: repos/web/src/routes/playground/+page.svelte [hunk 2]
```diff
@@ -1,5 +1,56 @@
 <script lang="ts">
+  import { onMount } from "svelte";
+
   let count = 0;
+  let subscribed;
+
+  onMount(async () => {
+    navigator.serviceWorker.ready.then(async (registration) => {
+      subscribed = await registration.pushManager.getSubscription();
+      return subscribed;
```
Comment (rob-3) [inline repos/web/src/routes/playground/+page.svelte:10]: blocking: If I understand correctly, I think this `return` is unnecessary.

**Review by rob-3 — CHANGES_REQUESTED** [review]
"suggestion:" means consider it, but no change is required.
"blocking:" means a change is required

**Review by rob-3 — CHANGES_REQUESTED** [review]

File: repos/web/src/routes/playground/+page.svelte [hunk 3]
```diff
@@ -1,5 +1,55 @@
...
+    });
+  });
+  const subscribeToPush = () => {
+    navigator.serviceWorker.ready
+      .then(async (registration) => {
+        subscribed = await registration.pushManager.subscribe({
+          userVisibleOnly: true,
+          applicationServerKey: import.meta.env.VITE_PUBLIC_VAPID_KEY
+        });
+        return subscribed;
```
Comment (rob-3) [inline repos/web/src/routes/playground/+page.svelte:19]: nit: you can actually do the assignment in the next `.then` to avoid this await

**Review by rob-3 — APPROVED** [review]

**Pull request merged**